### PR TITLE
[WIP] retry instructions for sake of incorrect order of execution

### PR DESCRIPTION
**Pull request by msarahan**
This is an implementation of a suggestion by @wulmer in https://github.com/conda/conda/issues/1811#issuecomment-159952754

The idea is to retry any failed instructions after the first pass.  Our dependency tree for actions is currently simple enough that this should work.  A better fix would be to figure out why installation is happening out of order with regards to dependencies.  This fix should be an OK intermediate.

This PR should not be merged without tests.

### CODE REVIEW

File: conda/instructions.py [hunk 1]
```diff
@@ -145,11 +148,29 @@ def execute_instructions(plan, index=None, verbose=False, _commands=None):
...
+            cmd(state, arg)
+        except:
+            failed_instructions.append((cmd, state, arg))
 
         if (state['i'] is not None and instruction in progress_cmds
                 and state['maxval'] == state['i']):
             state['i'] = None
             getLogger('progress.stop').info(None)
 
+    double_failures = []
```
Comment (wulmer) [inline conda/instructions.py]: Good idea to check for double failures! Just one idea about that:

This workaround is an iterative approach:
1. Run all pending commands.
2. If some commands fail, finish the others, then go to 1 with the failed commands.

The question here is when to stop this iteration. What about checking if the list of `failed_instructions` did not change? If there was no change during the last iteration step, it doesn't make sense to continue the iteration.

**Comment by ilanschnell** [comment]
Why should instructions fail in the first place?  And if the instructions fail, why would they then suddenly in the second pass not fail anymore?

**Comment by msarahan** [comment]
One example might be the order of installation being in alphabetical order rather than dependency order. If Qt is not present when pyqt is installed, the pyqt post link script will fail.

One other example might be some disk access fluke.  As I understand it, some of the OS module calls are asynchronous, and people occasionally report failures that later succeed.

Fixing the order of installation should mostly obviate this PR.  Fixing synchronicity somehow would also help.

To be perfectly clear, this PR is a quick band aid, and if you see better ways, I'm wide open to them.

**Comment by ilanschnell** [comment]
@msarahan Thanks for the response.  My position on this subject is best quoted by someone else:
```
When you must fail, fail noisily and as soon as possible.

   - ESR
```
Most likely, something went wrong for a reason, and to keep going will make things even worse.  The install order is something we have to fix anyway.
In most cases the user can simply rerun the conda command, when it failed.

**Comment by groutr** [comment]
How difficult is it to install dependencies in the correct order?

**Comment by ilanschnell** [comment]
Conda used to install dependencies in the correct order.  Sean made that change, but somehow it broke again.

**Pull request closed without merging**